packages/headless/src/api/commerce/product-listings/products-request.ts

```diff
@@ -0,0 +1,34 @@
+import {ProductRecommendation} from '../../search/search/product-recommendation';
+import {
+  baseProductListingRequest,
+  ProductListingsParam,
+} from './product-listings-params';
+
+export type GetProductsRequest = ProductListingsParam;
+
+export const buildGetProductsRequest = (req: GetProductsRequest) => {
+  const queryStringArguments: Record<string, string> = {};
+
+  return {
+    ...baseProductListingRequest(
+      req,
+      'POST',
+      'application/json',
+      queryStringArguments
+    ),
+    requestParams: prepareRequestParams(req),
+  };
+};
+
+const prepareRequestParams = (req: GetProductsRequest) => ({
+  url: req.url,
+});
+
+/**
+ * Defines the content of a successful response from the `/commerce/products/listing` API call.
+ */
+export interface GetProductsResponse {
```

Comment:
Similar to the above comment, maybe `ProductListingSuccessResponse` would be more consistent with the other use-cases?